src/main/java/com/minecolonies/coremod/client/gui/WindowHutGuardTower.java

```diff
@@ -9,255 +9,484 @@
-     */
-    private AbstractBuildingGuards.Task     task                = AbstractBuildingGuards.Task.GUARD;
-    /**
-     * The job of the guard, following the GuardJob enum.
-     */
-    private AbstractBuildingGuards.GuardJob job                 = null;
-    /**
-     * The list of MANUAL patrol targets.
-     */
-    private List<BlockPos>                  patrolTargets       = new ArrayList<>();
+    private List<BlockPos> patrolTargets = new ArrayList<>();
+
     /**
-     * The patrol list.
+     * The Map of mobs we are allowed to attack.
      */
-    private ScrollingList patrolList;
+    private List<MobEntryView> mobsToAttack = new ArrayList<>();
 
     /**
-     * Constructor for the window of the guardTower hut.
+     * Constructor for the window of the worker building.
      *
-     * @param building {@link AbstractBuildingGuards.View}.
+     * @param building class extending {@link AbstractBuildingGuards.View}.
      */
     public WindowHutGuardTower(final AbstractBuildingGuards.View building)
     {
-        super(building, Constants.MOD_ID + HUT_GUARD_TOWER_RESOURCE_SUFFIX);
+        super(building, Constants.MOD_ID + GUI_RESOURCE);
+
+        registerButton(GUI_BUTTON_JOB, this::switchJob);
+        registerButton(GUI_BUTTON_ASSIGNMENT_MODE, this::switchAssignmentMode);
+        registerButton(GUI_BUTTON_PATROL_MODE, this::switchPatrolMode);
+        registerButton(GUI_BUTTON_RETRIEVAL_MODE, this::switchRetrievalMode);
+        registerButton(GUI_BUTTON_SET_TARGET, this::setTarget);
+        registerButton(GUI_BUTTON_RECALCULATE, this::recalculate);
+
+        registerButton(GUI_SWITCH_TASK_PATROL, this::switchTask);
+        registerButton(GUI_SWITCH_TASK_FOLLOW, this::switchTask);
+        registerButton(GUI_SWITCH_TASK_GUARD, this::switchTask);
+
+        registerButton(GUI_LIST_BUTTON_SWITCH, this::switchAttackMode);
+        registerButton(GUI_LIST_BUTTON_UP, this::updatePriority);
+        registerButton(GUI_LIST_BUTTON_DOWN, this::updatePriority);
+
+        buttonTaskPatrol = this.findPaneOfTypeByID(GUI_SWITCH_TASK_PATROL, Button.class);
+        buttonTaskFollow = this.findPaneOfTypeByID(GUI_SWITCH_TASK_FOLLOW, Button.class);
+        buttonTaskGuard = this.findPaneOfTypeByID(GUI_SWITCH_TASK_GUARD, Button.class);
+        buttonSetTarget = this.findPaneOfTypeByID(GUI_BUTTON_SET_TARGET, Button.class);
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
 
         pullInfoFromHut();
 
-        registerButton(BUTTON_JOB, this::switchJob);
-        registerButton(BUTTON_ASSIGNMENT_MODE, this::switchAssignmentMode);
-        registerButton(BUTTON_PATROL_MODE, this::switchPatrolMode);
-        registerButton(BUTTON_RETRIEVAL_MODE, this::switchRetrievalMode);
+        patrolList = findPaneOfTypeByID(GUI_ELEMENT_LIST_LEVELS, ScrollingList.class);
+        if (task.equals(GuardTask.PATROL))
+        {
+            patrolList.setDataProvider(new ScrollingList.DataProvider()
```

Comment:
then better don't call it patrollist, call it "listOfPoints" or something like that.

Reply:
Not sure why, but Done